### PR TITLE
refactor(store): reduce @ngxs/store/internals bundlesize

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?

Reduced 0.06Kb

### CODE REVIEW

File: packages/store/src/internal/internals.ts [hunk 1]
```diff
@@ -380,3 +380,7 @@ export function getStateDiffChanges<T>(
   const currentValue: T = getValue(diff.newAppState, mappedStore.depth);
   return new NgxsSimpleChange(previousValue, currentValue, !mappedStore.isInitialised);
 }
+
+export function mergeObjects(a: object, b: object): object {
+  return { ...(a || {}), ...(b || {}) };
```
Comment (arturovt) [inline packages/store/src/internal/internals.ts]:
```suggestion
  return { ...a, ...b };
```

File: packages/store/src/internal/internals.ts [hunk 2]
```diff
@@ -380,3 +380,7 @@ export function getStateDiffChanges<T>(
   const currentValue: T = getValue(diff.newAppState, mappedStore.depth);
   return new NgxsSimpleChange(previousValue, currentValue, !mappedStore.isInitialised);
 }
+
+export function mergeObjects(a: object, b: object): object {
```
Comment (arturovt) [inline packages/store/src/internal/internals.ts]:
```suggestion
export function merge(a: object, b: object): object {
```

**Comment by arturovt** [comment]
This might be a refactor PR and not a fix.

**Comment by markwhitfeld** [comment]
Did we reduce the bundle by 6kB or by 0.06kB with this change?
I'm wondering if it is worth it?

**Comment by splincode** [comment]
@markwhitfeld 0.06kB
but improved readability and treeshaking :)

**Comment by markwhitfeld** [comment]
There doesn't really seem to be any real value to this refactor.
Was the merge function used by anything other than the store lib?

I'm actually wondering what the benefit of having the merge function was in the first place...
The only benefit would be the fact that all the extra code that ES2017 and earlier transpilation adds  (ie. `Object.assign(Object.assign({}, a), b)`) would now only be found in one place. Once you start looking at this it is a slippery slope because the normal spread syntax can be optimized by the V8 engine (I think) so then if you optimize for old you end up sacrificing the new.

**Comment by splincode** [comment]
@artuvotr

**Comment by arturovt** [comment]
Hey, Mr. Mark and Mr. Max :)

> the normal spread syntax can be optimized by the V8 engine

No. The TC39 spec says that spread operator (specifically for objects) is syntactic sugar over "copy data properties" from one object to the second one and evaluate assignment expression.

Spread for objects is a part of ES2018 spec which is slightly supported by modern browsers because it also has other features.

As we can't target ES2018 then this still will be transformed into `Object.assign` by the TS compiler.

> I'm actually wondering what the benefit of having the merge function was in the first place...

Yeah. I think we shouldn't have the `merge` function. I'm sure we can do this directly there:
```js
{ ...this._config.defaultsState, ...initialStateValue }
```

**Comment by arturovt** [comment]
I actually liked this PR as I approved it :slightly_smiling_face: Just thought that we could get rid of `merge` and move destruction there :)

**Pull request closed without merging**